spaghetti/embedding/AllMiniLMEmbedder.py

```diff
@@ -0,0 +1,86 @@
+"""
+Copyright 2023 Spaghetti team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+language governing permissions and limitations under the License.
+"""
+
+from transformers import AutoTokenizer, AutoModel
+import torch
+import torch.nn.functional as F
+import numpy as np
+
+from spaghetti.embedding.Embedder import Embedder
+
+class AllMiniLMEmbedder(Embedder):
+    """
+    Embedder for all-MiniLM-L6-v2 model. This class is responsible for loading the model and tokenizer,
+    and performing mean pooling on the model's output to generate sentence embeddings.
+    """
+
+    def __init__(self):
+        self._model = self._init_model()
+        self._tokenizer = self._init_tokenizer()
+
+    def _init_model(self):
+        """
+        This method initializes the model
+        """
+        return AutoModel.from_pretrained('sentence-transformers/all-MiniLM-L6-v2')
+
+    def _init_tokenizer(self):
+        """
+        This method initializes the tokenizer
+        """
+        return AutoTokenizer.from_pretrained('sentence-transformers/all-MiniLM-L6-v2')
+
+    def embed(self, data: str) -> np.array:
+        """
+        This method should embed the input data
+        :param data: Input data to be embedded
+        :return: Embedded data
+        """
+        # Tokenize the input data
+        inputs = self._tokenizer(data, return_tensors='pt', truncation=True, padding=True)
+
+        # Get the model's output
+        model_output = self._model(**inputs)
+
+        # Perform mean pooling on the model's output to generate sentence embeddings
+        embeddings = self._mean_pooling(model_output, inputs['attention_mask'])
+
+        return embeddings.detach().numpy()
+
+    def embed_batch(self, data: list[str]) -> np.array:
```

Comment:
It's better to use `typing.List` for type hints with list. And it's better to use `np.ndarray` for type hints, `np.array` is a function and `np.ndarray` is a class